src/pybind/mgr/dashboard/frontend/src/app/shared/services/data-gateway.service.ts

```diff
@@ -26,17 +27,23 @@ export class DataGatewayService {
     return this.cache[cacheable];
   }
 
-  create(dataPath: string, data: any): Observable<any> {
+  request(requestMethod: string, dataPath: string, data: any): Observable<any> {
     const match = dataPath.match(/(?<url>[^@]+)(?:@(?<version>.+))?/);
     const url = match.groups.url.split('.').join('/');
     const version = match.groups.version || '1.0';
-
-    return this.http.post<any>(url, data, {
-      headers: { Accept: `application/vnd.ceph.api.v${version}+json` }
-    });
+    const headers = { Accept: `application/vnd.ceph.api.v${version}+json` };
+    return this.http[requestMethod]<any>(url, data, { headers });
```

Comment:
```suggestion
    return this.http<any>.request(requestMethod, url, data, { headers });
```
I've seen several posts where people use indexing as a way to dynamically change the request type, but it seems like the [documented](https://angular.io/api/common/http/HttpClient#request) way in angular is `http.request(method_type, ...`